api/client/client.go

```diff
@@ -3126,6 +3146,84 @@ type ResourcePage[T types.ResourceWithLabels] struct {
 	NextKey string
 }
 
+// getTypeFromUnifiedResource is used to extract the resource kind from the PaginatedResource returned
+// from the rpc ListUnifiedResources
+func getTypeFromUnifiedResource(resource *proto.PaginatedResource) (types.ResourceWithLabels, error) {
```

Comment:
i found an example of this elsewhere in the codebase but I don't like it. although, the other type casting that i use in the rest of the pr didn't work. happy for feedback here